### PR TITLE
Extract steps of submitPayment() so it's easier to follow

### PR DESCRIPTION
**Pull request by jeremywiebe**
This PR just refactors the `submitPayment` command in the SFCC connector so the steps are a bit easier to follow. 

 **JIRA**: (link to JIRA ticket)
 **Linked PRs**: (links to corresponding PRs, optional)

## Changes
- Extract the steps in `submitPayment` into individual functions so it's easier to see the overall flow

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions SFCC](https://preview.mobify.com/?url=https%3A%2F%2Fmobify-tech-prtnr-na03-dw.demandware.net%2Fon%2Fdemandware.store%2FSites-2017refresh-Site%2Fdefault%2FHome-Show&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- Run all the way through checkout

### CODE REVIEW

File: web/app/integration-manager/_sfcc-connector/checkout/commands.js [hunk 1]
```diff
@@ -150,98 +150,110 @@ export const submitShipping = (formValues) => (dispatch) => {
...
-                return makeSfccRequest(`${API_END_POINT_URL}/baskets/${basket.basket_id}/billing_address?use_as_shipping=false`, requestOptions)
-                    .then((response) => response.json())
-                    .then((responseJSON) => {
-                        if (responseJSON.fault) {
-                            throw new Error(responseJSON.fault.message)
-                        }
-                        return responseJSON
-                    })
+                return responseJSON /* OCAPI Basket */
+            })
```
Comment (DanielBrookRoberge) [inline web/app/integration-manager/_sfcc-connector/checkout/commands.js]: We probably want to make a new utility `makeSfccJsonRequest` which encapsulates all of this (including the `JSON.stringify` of the body, probably). This might be a separate PR.

File: web/app/integration-manager/_sfcc-connector/checkout/commands.js [hunk 2]
```diff
@@ -150,98 +150,110 @@ export const submitShipping = (formValues) => (dispatch) => {
...
+    const type = getCardData(formValues.ccnumber).cardType
+    const expiryMonth = /^\d\d/.exec(formValues.ccexpiry)[0]
+    const expiryYear = /\d\d$/.exec(formValues.ccexpiry)[0]
+    const paymentInstrumentID = order.payment_instruments[0].payment_instrument_id
+    const requestOptions = {
+        method: 'PATCH',
+        body: JSON.stringify({
+            payment_card: {
+                card_type: type,
+                expiration_month: parseInt(expiryMonth),
```
Comment (DanielBrookRoberge) [inline web/app/integration-manager/_sfcc-connector/checkout/commands.js]: Shouldn't this be `parseInt(expiryMonth, 10)` since we're expecting leading 0's?

Comment (jeremywiebe) [inline web/app/integration-manager/_sfcc-connector/checkout/commands.js]: I'm not certain.  I didn't change this line when I moved it. I'll test with non-double-digit months and see.

File: web/app/integration-manager/_sfcc-connector/checkout/commands.js (hunk 1 above)
Comment (jeremywiebe) [inline web/app/integration-manager/_sfcc-connector/checkout/commands.js]: I've had that thought rolling around in my mind for a while. :)  I think I'll opt for another PR.

File: web/app/integration-manager/_sfcc-connector/checkout/commands.js (hunk 2 above)
Comment (jeremywiebe) [inline web/app/integration-manager/_sfcc-connector/checkout/commands.js]: So Demandware seems to accept a single-digit month. Or are you fearing that things will fall apart when there's an actual payment processor?

image

Comment (DanielBrookRoberge) [inline web/app/integration-manager/_sfcc-connector/checkout/commands.js]: Looking at the docs for `parseInt` we don't support browsers with the old behaviour for leading zeros. So this is fine.
